kolibri/core/content/test/test_content_sync_hook.py

```diff
@@ -0,0 +1,48 @@
+import mock
+from morango.sync.context import SessionContext
+
+from kolibri.core.auth.sync_operations import KolibriSyncOperations
+from kolibri.core.content.kolibri_plugin import ContentSyncHook
+from kolibri.core.content.test.utils.test_content_request import (
+    IncompleteDownloadsQuerysetTestCase,
+)
+from kolibri.core.device.models import DeviceStatus
+
+
+class KolibriContentSyncHookTestCase(IncompleteDownloadsQuerysetTestCase):
+    def setUp(self):
+        super(KolibriContentSyncHookTestCase, self).setUp()
+        self.operation = KolibriSyncOperations()
+        self.context = mock.Mock(spec_set=SessionContext)()
+
+    @mock.patch("kolibri.core.device.models.LearnerDeviceStatus.save_learner_status")
+    @mock.patch("kolibri.core.content.utils.content_request.StorageCalculator")
+    @mock.patch("kolibri.core.content.kolibri_plugin.ContentSyncHook")
+    def test_post_transfer_sets_insufficient_storage(
+        self,
+        mock_hook,
+        mock_calc,
+        mock_save_learner_status,
+    ):
+        with mock.patch(
+            "kolibri.core.content.utils.settings.automatic_download_enabled",
+            return_value=True,
+        ):
+            with mock.patch(
+                "kolibri.core.content.utils.content_request.get_free_space_for_downloads",
+                return_value=0,
+            ):
+                self._create_resources(self.admin_request.contentnode_id)
+                mock_hook.post_transfer = lambda *args: ContentSyncHook.post_transfer(
+                    mock_hook, *args
+                )
```

Comment:
Looks like python2.7 tests are failing on this test. There shouldn't be any harm AFAIK in creating an instance of the `ContentSyncHook` so you could try to remove the mocking.